core/src/test/java/com/linecorp/armeria/client/TestReconfigurableServer.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+
+class TestReconfigurableServer {
+
+    @Test
+    public void test_reconfiguration_of_server() throws Exception {
+        final ServerBuilder sb = Server.builder();
+        sb.http(8080);
+
+        sb.service("/test1", (ctx, req) -> {
+            return HttpResponse.of("Hello, world!");
+        });
+
+        sb.service("/test11/{name}", (ctx, req) -> {
+            final String param = ctx.pathParam("name");
+            return HttpResponse.of("Hello, " + param.toUpperCase());
+        });
+
+        final Server server = sb.build();
+
+        final CompletableFuture<Void> future = server.start();
+        Thread.sleep(1000);
+        try (CloseableHttpClient hc = HttpClients.createMinimal()) {
```

Comment:
How about using Armeria `WebClient`?
We will remove the Apache client from the test code eventually.

Reply:
Yes will do.